Code/Mantid/Framework/Algorithms/src/CheckWorkspacesMatch.cpp

```diff
@@ -640,7 +641,14 @@ bool CheckWorkspacesMatch::checkAxes(API::MatrixWorkspace_const_sptr ws1,
     // Use Axis's equality operator to check length and values
     // Don't check spectra axis as that just takes it values from the ISpectrum
     // (see checkSpectraMap)
-    if (!ax1->isSpectra() && !ax1->operator==(*ax2)) {
+    if (ax1->isNumeric() && ax2->isNumeric()) {
+      const NumericAxis *na1 = static_cast<const NumericAxis *>(ax1);
+      const double tolerance = getProperty("Tolerance");
+      if (!na1->equalWithinTolerance(*ax2, tolerance)) {
+        result = axis_name + " values mismatch";
+        return false;
+      }
+    } else if (!ax1->isSpectra() && !ax1->operator==(*ax2)) {
```

Comment:
I think we are better off using the overloaded operator== instead of this first if statement. RefAxis and NumericAxis both return true for isNumeric(), but have different implementations of operator==.
  
The ISISIndirectAnalysis system test passes on os x 10.9 if we remove the first if.